partiql-plan/src/main/kotlin/org/partiql/plan/rex/RexVar.kt

```diff
@@ -42,15 +42,13 @@ internal class RexVarImpl(depth: Int, offset: Int, type: RexType) : RexVar {
         if (other !is RexVar) return false
         if (_depth != other.getDepth()) return false
         if (_offset != other.getOffset()) return false
-        if (_type != other.getType()) return false
         return true
     }
 
     override fun hashCode(): Int {
         var result = 1
         result = 31 * result + _depth
         result = 31 * result + _offset
-        result = 31 * result + _type.hashCode()
```

Comment:
At some point, need to consider whether `type` should or shouldn't be included in the `equals` and `hashcode` functions for the plan node classes.